### PR TITLE
SE050 crypto driver: add support for SE050-F

### PR DESCRIPTION
**Pull request by ldts**
The SE050-F has been FIPS 140-2 certified with Security Level 3 for OS and Applet, and  Security Level 4 related to Physical Security of the HW. 

The SE050-F requires a specific configuration according to the certification.

Expected boot log.

```
I/TC: Non-secure external DT found                                                                                                                    
I/TC: OP-TEE version: 3.19.0-26-g7ba53da49 (gcc version 11.3.0 (GCC)) #1 Wed Oct 26 08:53:14 UTC 2022 aarch64                                         
I/TC: WARNING: This OP-TEE configuration might be insecure!                                                                                           
I/TC: WARNING: Please check https://optee.readthedocs.io/en/latest/architecture/porting_guidelines.html                                               
I/TC: Primary CPU initializing                                                                                                                        
I/TC: se050: sss: SCP03 enabled                                                                                                                       
I/TC: se050: Info: Applet ID                                                                                                                          
I/TC: se050: Info:      04.00.50.01.9f.05.14.b9 52.16.9a.04.2f.0b.69.b6                                                                               
I/TC: se050: Info:      00.00                                                                                                                         
I/TC: se050: Info: Applet Major = 3                                                                                                                  
I/TC: se050: Info: Applet Minor = 6                                                                                                                  
I/TC: se050: Info: Applet patch = 0                                                                                                                   
I/TC: se050: Info: AppletConfig = 61D2                                                                                                                
I/TC: se050: Info:      without ECDAA                                                                                                                 
I/TC: se050: Info:      with    ECDSA_ECDH_ECDHE                                                                                                      
I/TC: se050: Info:      without EDDSA                                                                                                                 
I/TC: se050: Info:      without DH_MONT                                                                                                               
I/TC: se050: Info:      with    HMAC                                                                                                                  
I/TC: se050: Info:      without RSA_PLAIN                                                                                                             
I/TC: se050: Info:      with    RSA_CRT                                                                                                               
I/TC: se050: Info:      with    AES                                                                                                                   
I/TC: se050: Info:      with    DES                                                                                                                   
I/TC: se050: Info:      without PBKDF                                                                                                                 
I/TC: se050: Info:      without TLS
I/TC: se050: Info:      without MIFARE
I/TC: se050: Info:      with    I2CM
I/TC: se050: Info: Internal = 010B
I/TC: se050: Info: SE050 JCOP4 Information:
I/TC: se050: Info: Tag value - proprietary data 0xFE = 0xFE
I/TC: se050: Info: Length of following data 0x45 = 0x45
I/TC: se050: Info: Tag card identification data
I/TC: se050: Info:      df.28
I/TC: se050: Info: Length of card identification data = 0x42
I/TC: se050: Info: Tag configuration ID (Must be 0x01) = 0x01
I/TC: se050: Info: Length configuration ID 0x0C = 0x0C
I/TC: se050: Info: Configuration ID
I/TC: se050: Info:      00.01.a9.2a.1a.63.58.ee b6.d7.73.c0
I/TC: se050: Info: OEF ID            
I/TC: se050: Info:      a9.2a
I/TC: se050: Info: Tag patch ID (Must be 0x02) = 0x02
I/TC: se050: Info: Length patch ID 0x08 = 0x08
I/TC: se050: Info: Patch ID
I/TC: se050: Info:      00.00.00.00.00.00.00.01 
I/TC: se050: Info: Tag platform build ID1 (Must be 0x03) = 0x03
I/TC: se050: Info: Length platform build ID 0x18 = 0x18
I/TC: se050: Info: Platform build ID
I/TC: se050: Info:      4a.33.52.33.35.31.30.32 36.34.35.37.31.31.30.30
I/TC: se050: Info:      03.4d.67.74.0b.e1.42.19 
I/TC: se050: Info: JCOP Platform ID = J3R3510264571100
I/TC: se050: Info: Tag FIPS mode (Must be 0x05) = 0x05
I/TC: se050: Info: Length FIPS mode 0x01 = 0x01
I/TC: se050: Info: FIPS mode var = 0x01
I/TC: se050: Info: Tag pre-perso state (Must be 0x07) = 0x07
I/TC: se050: Info: Length pre-perso state 0x01 = 0x01
I/TC: se050: Info: Bit mask of pre-perso state var = 0x00
I/TC: se050: Info: Tag ROM ID (Must be 0x08) = 0x08
I/TC: se050: Info: Length ROM ID 0x08 = 0x08
I/TC: se050: Info: ROM ID            
I/TC: se050: Info:      2e.5a.d8.84.09.c9.ba.db 
I/TC: se050: Info: Status Word (SW)
I/TC: se050: Info:      90.00
I/TC: se050: sss: SCP03 enabled
```

### CODE REVIEW

**Comment by jforissier** [comment]
Comment on "crypto: se050: SCP03 fix SE050F2 keys, add model SE050F"

> The SE050F keys were incorrectly assigned to SE050F2.

Isn't it the other way around? (i.e., SE050F2 were incorrectly assigned to SE050F). The SE050F keys seem to be introduced in this PR so they could not be assigned to anything before, right? I suppose SE050F2 was just not configured since its ID was incorrect?

Could this commit be split in two, one to fix SE050F2_ID, then the rest to support SE050F?

File: core/drivers/crypto/se050/session.c [hunk 1]
```diff
@@ -64,32 +68,62 @@ static TEE_Result enable_scp03(void)
...
+	return TEE_SUCCESS;
+}
+
+static TEE_Result se050_early_init_scp03(void)
+{
+	/* Initialize session */
+	se050_session = (sss_se05x_session_t *)((void *)&se050_ctx.session);
+	se050_kstore = (sss_se05x_key_store_t *)((void *)&se050_ctx.ks);
+
+#ifdef CFG_CORE_SE05X_OEFID
```
Comment (jforissier) [inline core/drivers/crypto/se050/session.c:99]: `if IS_ENABLED(CFG_CORE_SE05X_OEFID) {`

**Review by jforissier — COMMENTED** [review]
For "crypto: se050: SCP03 enabled only session.", one minor comment, then:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/drivers/crypto/se050/session.c (hunk 1 above)
Comment (ldts) [reply core/drivers/crypto/se050/session.c:99]: um, does this apply to strings as well? ie, will I have to define a default?

Comment (jforissier) [reply core/drivers/crypto/se050/session.c:99]: No, you're right. We would need a special macro for this. The `#if` is fine then.

**Comment by ldts** [comment]
> Comment on "crypto: se050: SCP03 fix SE050F2 keys, add model SE050F"
> 
> > The SE050F keys were incorrectly assigned to SE050F2.
> 
> Isn't it the other way around? (i.e., SE050F2 were incorrectly assigned to SE050F). The SE050F keys seem to be introduced in this PR so they could not be assigned to anything before, right? I suppose SE050F2 was just not configured since its ID was incorrect?

right, I'' be more clear with the commit message and split it in two

> 
> Could this commit be split in two, one to fix SE050F2_ID, then the rest to support SE050F?

yep

**Comment by ldts** [comment]
tags added

**Pull request merged**